### ISSUE TITLE
Creates null c_cpp_properties.json when running cmake: Configure

### ISSUE THREAD

**Issue by DaOnlyOwner** [maddouri/vscode-cmake-tools-helper#10]
Hello ;)
The title says it all: I run cmake: Configure and this extension creates a "null" .json file (no paths etc. specified.) 
Cmake finishes without any errors. Any idea?
Cheers,
DaOnlyOwner

**Comment by inter1965** [comment]
I witness the same problem.

**Comment by pacxx** [comment]
Same for me.

**Comment by DaOnlyOwner** [comment]
@pacxx And it resolved the issue?

**Comment by pacxx** [comment]
No, just let me use vscode with my include path.

**Comment by pacxx** [comment]
I investigated the problem a little bit and it seems the problem starts in line 49 of state.ts const activeTargetName = cmakeToolsWrapper.defaultBuildTarget;
defaultBuildTarget is null.

**Comment by pacxx** [comment]
Ok is investigated further and I think the problem is not this extension the cmake-server does not report the includePath value and cmake-tools-helper just founds empty (or better said undefined) includePath arrays.

**Comment by DaOnlyOwner** [comment]
Uh thats kinda annoying. Did you find a way to fix this problem?

**Comment by LeonineKing1199** [comment]
Yeah, I too am experiencing this issue. I've just disabled the extension and will probably just give it like 1 star.

**Comment by bietje** [comment]
Is there any progress on this issue?

**Comment by gabyx** [comment]
same issue

**Comment by batuakan** [comment]
Same here,
Mostly it doesnt create the c_cpp_properties.json file, and when it does it is null as described above

**Comment by koguchid** [comment]
Same!

**Comment by h00shi** [comment]
Same here. I was really hoping to migrate from QtCreator.

**Comment by getbyte** [comment]
I have that problem too. IncludePath in c_cpp_properties.json was cleaned exactly after I pressed "Build". I tried to rebuild project a few times, and I found out that file didn't changed when I press "Build" again, even when I'd clean CMakeFiles/CMakeTools folders, or executable results. But if only I'll delete CMakeCache.txt - oops... Problem is here again!

**Comment by DaOnlyOwner** [comment]
I just disabled it and used the default settings to find the std libs.

**Comment by h00shi** [comment]
@DaOnlyOwner Yes I used the same strategy.

**Comment by Randshot** [comment]
The problem lies in having "all" selected as build target. You have to specify a specific build target for it to work.

**Comment by tim37021** [comment]
@Randshot thx, this fix my problem. however, it does not add standard header into includePath..

**Comment by Randshot** [comment]
@tim37021 I know. This is a known problem with the current extension and as far as I know, there is currently no easy way to get these standard include headers.

**Comment by tim37021** [comment]
@Randshot But somehow cpptools is able to get these standard include headers in my computer. Maybe they have the solution?
